src/vm/codeversion.cpp

```diff
@@ -2135,6 +2138,12 @@ PCODE CodeVersionManager::PublishVersionableCodeIfNecessary(MethodDesc* pMethodD
         pCode = activeVersion.GetNativeCode();
         if (pCode == NULL)
         {
+            if (pTieredCompilationManager != nullptr &&
```

Comment:
As we chatted about, lets move this notification inside the PrepareCode call path to a spot where we know we really are jitting. Calling from here includes all R2R and multicore jit cases that may not JIT but misses non-versionable code that is jitting.

Reply:
Oh missed that, will do